conjure-go-client/httpclient/metrics.go

```diff
@@ -175,3 +182,77 @@ func tlsVersionString(version uint16) string {
 	}
 	return ""
 }
+
+// dialer is the interface implemented by net.Dialer and proxy.Dialer
+type dialer interface {
+	Dial(network, addr string) (c net.Conn, err error)
+}
+
+// dialer is the newer interface implemented by net.Dialer and proxy.Dialer
+type contextDialer interface {
+	DialContext(ctx context.Context, network, addr string) (c net.Conn, err error)
+}
+
+// metricsWrappedConn is a wrapper for net.Dialer that tracks a metric of in-flight connections.
+type metricsWrappedDialer struct {
+	dialer dialer
+}
+
+func (d *metricsWrappedDialer) DialContext(ctx context.Context, network, addr string) (net.Conn, error) {
+	var conn net.Conn
+	var err error
+	if cDialer, ok := d.dialer.(contextDialer); ok {
+		conn, err = cDialer.DialContext(ctx, network, addr)
+	} else {
+		conn, err = d.dialer.Dial(network, addr)
+	}
+	if err != nil {
+		return nil, err
+	}
+	counter := metrics.FromContext(ctx).Counter(MetricInFlightConns)
```

Comment:
we could add the addr here as a tag if we want to so we know where the connections are going... thoughts?

Reply:
It would have pretty high fanout, so we don't record the address in the java metrics. We do include additional data in a debug level log line in our instrumentation though, perhaps something similar would be a good fit here?

Reply:
yeah agreed, I decided to split the difference and add the service-name to match the other metrics